src/Compilers/Core/Portable/Syntax/GreenNode.cs

```diff
@@ -730,7 +742,11 @@ public virtual GreenNode WithTrailingTrivia(GreenNode? trivia)
                     }
                 }
                 node = firstChild;
-            } while (node?._slotCount > 0);
+            }
+            // Note: it's ok to examine SmallSlotCount here.  All we're trying to do is make sure we have at least one
+            // child.  And SmallSlotCount works both for small counts and large counts.  This avoids an unnecessary
+            // virtual call for large list nodes.
+            while (node?._nodeFlagsAndSlotCount.SmallSlotCount > 0);
```

Comment:
this code was def odd (the direct use of _slotCount).  but it's technically totally ok to do that.  i've added docs to say why.